django-verdant/rca/api/filters.py

```diff
@@ -0,0 +1,40 @@
+from datetime import datetime
+from rest_framework import filters
+
+from django.core.exceptions import FieldDoesNotExist
+from django.db.models.fields.related import ForeignObjectRel, OneToOneRel
+from rca.models import EventItemDatesTimes, EventItem
+
+class RelatedProgrammesFilter(filters.BaseFilterBackend):
+    def filter_queryset(self, request, queryset, view):
+        if hasattr(queryset.model, 'related_programmes'):
+            rp = request.GET.getlist('rp', [])
+            if rp:
+                queryset = queryset.filter(related_programmes__programme__slug__in=rp)
+
+        if hasattr(queryset.model, 'related_programme'):
+            rp = request.GET.getlist('rp', [])
+            if rp:
+                queryset = queryset.filter(related_programme__slug__in=rp)
+
+        if hasattr(queryset.model, 'roles'):
+            rp = request.GET.getlist('rp', [])
+            if rp:
+                queryset = queryset.filter(roles__programme__slug__in=rp)
```

Comment:
What if a model has two or three of the fields? Is that possible? Then probably results may be zero. Don't we need to have OR logic between these three filters?

Reply:
Discussed over slack. Models have one of these fields, rather than a potential mixture of them.